operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -146,9 +151,206 @@ private void addInitContainer(Deployment baseDeployment, List<String> extensions
         baseDeployment.getSpec().getTemplate().getSpec().setInitContainers(Collections.singletonList(initContainer));
     }
 
+    public void validatePodTemplate(KeycloakStatusBuilder status) {
+        if (keycloakCR.getSpec() == null ||
+                keycloakCR.getSpec().getUnsupported() == null ||
+                keycloakCR.getSpec().getUnsupported().getPodTemplate() == null) {
+            return;
+        }
+        var overlayTemplate = this.keycloakCR.getSpec().getUnsupported().getPodTemplate();
+
+        if (overlayTemplate.getMetadata() != null &&
+            overlayTemplate.getMetadata().getName() != null) {
+            status.addWarningMessage("The name of the podTemplate cannot be modified");
+        }
+
+        if (overlayTemplate.getMetadata() != null &&
+            overlayTemplate.getMetadata().getNamespace() != null) {
+            status.addWarningMessage("The namespace of the podTemplate cannot be modified");
+        }
+
+        if (overlayTemplate.getSpec() != null &&
+            overlayTemplate.getSpec().getContainers() != null &&
+            overlayTemplate.getSpec().getContainers().get(0) != null &&
+            overlayTemplate.getSpec().getContainers().get(0).getName() != null) {
+            status.addWarningMessage("The name of the keycloak container cannot be modified");
+        }
+    }
+
+    private <T, V> void mergeMaps(Map<T, V> map1, Map<T, V> map2, Consumer<Map<T, V>> consumer) {
+        var map = new HashMap<T, V>();
+        Optional.ofNullable(map1).ifPresent(e -> map.putAll(e));
+        Optional.ofNullable(map2).ifPresent(e -> map.putAll(e));
+        consumer.accept(map);
+    }
+
+    private <T> void mergeLists(List<T> list1, List<T> list2, Consumer<List<T>> consumer) {
+        var list = new ArrayList<T>();
+        Optional.ofNullable(list1).ifPresent(e -> list.addAll(e));
+        Optional.ofNullable(list2).ifPresent(e -> list.addAll(e));
+        consumer.accept(list);
+    }
+
+    private <T> void mergeField(T value, Consumer<T> consumer) {
+        Optional.ofNullable(value).ifPresent(v -> {
+            if (v != null && (!(v instanceof List) || (v instanceof List && ((List<?>) v).size() > 0))) {
```

Comment:
Do we really care about lists? Are there any implications if we set an empty list? We don't care in `mergeLists`.

Reply:
Yes, in this case we do, for correctly merging fields like `command` or `args` using as logic:
if it's defined in `podTemplate` should override the entire list (as opposed to merging it).

Reply:
Ok, makes sense. 👍  We do not support "undefining" fields, just overriding with a new value.